L/LibGit2/build_tarballs.jl

```diff
@@ -53,9 +56,9 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    "MbedTLS_jll",
-    "LibSSH2_jll",
-    "LibCURL_jll",
+    Dependency("Libiconv_jll"),
```

Comment:
@staticfloat since libgit2 is a dependency of Julia, maybe adding another binary dependency here is not desirable?

Reply:
We drop CURL so the number of deps doesn't change. :-) Why wasn't iconv needed before BTW?

Is there a way to have this dependency only on Musl? Apparently the system iconv is good enough on most other systems according to https://github.com/libgit2/libgit2/commit/2e2d8c6493ec4d151c55d7421c93126267ee8e6d.

Reply:
https://github.com/JuliaPackaging/Yggdrasil/pull/638#issuecomment-602718435

Reply:
It's not strictly needed, but we found that not linking to JLL libiconv can lead to nasty bugs at runtime in case the system libiconv is different from the expected one.

Reply:
Hmmmm.  IMO, let's not do this, since it hasn't been causing problems on MacOS so far.  I'd like to minimize the dependencies we rely on from Julia.

Reply:
So not use iconv anywhere? Do we support Musl (it seems to be needed there)?

Reply:
I think I added iconv only for macOS (but Linux wasn't working yet at that time)

Reply:
> it hasn't been causing problems on MacOS so far

For the record, I did have problems on Travis because of macOS libiconv: https://github.com/JuliaGtk/GtkSourceWidget.jl/pull/9#issuecomment-569546758.  It took me a while to debug that.

Reply:
What I mean is that the binaries so far haven't been causing issues with libiconv, and we've been linking against libiconv this whole time, right?